spock-core/src/main/java/org/spockframework/runtime/DataVariablesIterationNameProvider.java

```diff
@@ -51,7 +51,11 @@ public String getName(IterationInfo iteration) {
       dataVariables.forEach((name, value) -> {
         String valueString;
         try {
-          valueString = toStringOrDump(value);
+          valueString = RenderUtil.toStringOrDump(value);
+          int maxDataValueStringLength = 1024;
```

Comment:
❓ Did you try to figure out which length of method names where problematic?

❓ Does this really fix the problem comprehensively? What happens if you have more than 1 problematic parameter?